### PR TITLE
Improve performance for :meth:`~become`

### PR DESCRIPTION
**Pull request by Darylgolden**
## Overview: What does this pull request change?

- Added a boolean parameter, `skip_point_alignment` to the `align_data` method. When True, the expensive point alignment method is not called.
- Added documentation for the `align_data` method.
- Added three tests testing the `become` method.

## Motivation and Explanation: Why and how do your changes improve the library?
While profiling the following scene:

```py
class Test(Scene):
    def construct(self):
        nplane = NumberPlane()
        b = ValueTracker(1)
        a = always_redraw(
            lambda: nplane.plot_implicit_curve(
                lambda x, y: x ** 3 - y ** 2 - x - b.get_value()
            )
        )
        self.add(nplane, a)
        self.wait()
        self.play(b.animate.set_value(-1))
        self.wait()
```

I noticed that Manim was spending an inordinate amount of time on the `become` method of Mobject. In particular, when using the `become` method for VMobjects, a lot of time is wasted aligning the points of the VMobject to its target. This is despite the fact that the `become` method simply changes a `VMobject`'s point data to match its target, with no need to interpolate where its initial points should lie (that will immediately get overwritten). Skipping point alignment caused the render time for the above scene on medium quality on my PC to drop from 80 seconds to 15 seconds with Cairo. For OpenGL, the time spent on the become method was already negligible, so I left it untouched.

## Reviewer Checklist
- [ ] The PR title is descriptive enough for the changelog, and the PR is labeled correctly
- [ ] If applicable: newly added non-private functions and classes have a docstring including a short summary and a PARAMETERS section
- [ ] If applicable: newly added functions and classes are tested

### CODE REVIEW

**Comment by Darylgolden** [comment]
Docs build has dropped from ~980 seconds to ~930 seconds 💯

File: manim/mobject/mobject.py [hunk 1]
```diff
@@ -2471,10 +2471,11 @@ def construct(self):
         return self.shuffle(*args, **kwargs)
 
     # Alignment
-    def align_data(self, mobject: "Mobject"):
+    def align_data(self, mobject: "Mobject", skip_point_alignment: bool = False):
```
Comment (behackl) [inline manim/mobject/mobject.py:2474]: Suggestion for some bare-bones documentation of align_data

```suggestion
    def align_data(self, mobject: "Mobject", skip_point_alignment: bool = False):
    """Aligns the data of this mobject with another mobject.

    Afterwards, the two mobjects will have the same number of submobjects
    (see :meth:`.align_submobjects`), the same parent structure (see
    :meth:`.null_point_align`), and the same number of points
    (see :meth:`.align_points`).

    Parameters
    ----------
    mobject
        The other mobject this mobject should be aligned to.
    skip_point_alignment
        Controls whether or not the computationally expensive
        point alignment is skipped (default: False).
    """
```

**Review by behackl — APPROVED** [review]
LGTM! Left one suggestion for documentation, but other than that this looks good to me.

File: manim/mobject/mobject.py [hunk 2]
```diff
@@ -2471,10 +2471,26 @@ def construct(self):
...
+    """Aligns the data of this mobject with another mobject.
+
+    Afterwards, the two mobjects will have the same number of submobjects
+    (see :meth:`.align_submobjects`), the same parent structure (see
+    :meth:`.null_point_align`), and the same number of points
+    (see :meth:`.align_points`).
```
Comment (Darylgolden) [inline manim/mobject/mobject.py:2480]:
```suggestion
    :meth:`.null_point_align`). If `skip_point_alignment` is false,
    they will also have the same number of points (see :meth:`.align_points`).
```

Comment (behackl) [reply manim/mobject/mobject.py]: Code literals have two backticks.

Comment (Darylgolden) [reply manim/mobject/mobject.py]: oops :sweat:

**Pull request merged**